NanoCORE/DiPhotonSelections.h

```diff
@@ -26,8 +26,9 @@ struct Photon {
         //trkSumPtHollowConeDR03_ = nt.Photon_trkSumPtHollowConeDR03()[idx_]; // To be readded
         //idlevel_ = whichPhotonLevel(id_, idx_);
         try { fixedGridRhoFastjetAll_ = nt.Rho_fixedGridRhoFastjetAll(); } catch(const std::exception& e) { fixedGridRhoFastjetAll_ = nt.fixedGridRhoFastjetAll(); }
+        try { genPartFlav_ = nt.Photon_genPartFlav()[idx_]; } catch(const std::exception& e) { genPartFlav_ = -1; }
     }
-    void setGenPartFlav(unsigned int idx) { genPartFlav_ = nt.Photon_genPartFlav()[idx_]; }
+//    void setGenPartFlav(unsigned int idx) { genPartFlav_ = nt.Photon_genPartFlav()[idx_]; }
```

Comment:
We comment out this function, which is ok, since it is no longer needed but then we still use it on L415 (L386 of the main branch). How is this possible? Has this PR been tested? If so, how does it work? When testing, did you recompile NanoCORE?

By the way, if this is really not needed, why not delete it?

Reply:
Yes, this function shouldn't be used anymore in the ScanChain_Hgg.C . L438 and L439 has already get the photon genPartFlav variable